src/analyses/goto_check.cpp

```diff
@@ -1880,7 +1917,13 @@ void goto_checkt::goto_check(
     {
       const code_assignt &code_assign=to_code_assign(i.code);
 
-      check(code_assign.lhs());
+      // Reset the no_enum_check with the flag reset for exception
+      // safety
+      {
+        flag_resett no_enum_check_flag_resetter;
+        no_enum_check_flag_resetter.set_flag(no_enum_check, true);
+        check(code_assign.lhs());
+      }
```

Comment:
Placing this comment at a random point: can you please also extend `ansi-c/scanner.l` to recognize enum-range as part of `named_checks` and add a `disable:` case around line 1800 in this file here (`analyses/goto_check.cpp`).

Reply:
That's a good point, thanks